tests/testthat/test-transform_pmf_model_out.R

```diff
@@ -0,0 +1,67 @@
+# This file uses test fixture data in the following three objects:
+# tests/testthat/testdata/pmf_test_model_out_tbl.rds
+# tests/testthat/testdata/pmf_test_target_observations.rds
+# tests/testthat/testdata/pmf_test_expected_merged_forecast.rds
+#
+# These files are constructed "by hand" in tests/testthat/testdata/make_pmf_test_data.R
+# By construction, model_out_tbl and target_observations are valid inputs to transform_pmf_model_out,
+# and exp_forecast is the expected return value from transform_pmf_model_out
+
+test_that("transform_pmf_model_out succeeds with valid inputs", {
+  model_out_tbl <- readRDS(test_path("testdata", "pmf_test_model_out_tbl.rds"))
+  target_observations <- readRDS(test_path("testdata", "pmf_test_target_observations.rds"))
+  exp_forecast <- readRDS(test_path("testdata", "pmf_test_expected_merged_forecast.rds"))
+
+  act_forecast <- transform_pmf_model_out(
+    model_out_tbl = model_out_tbl,
+    target_observations = target_observations
+  )
+
+  expect_equal(act_forecast, exp_forecast, ignore_attr = "groups")
+})
+
```

Comment:
If we want to keep the `output_type_id_order` argument, this will act as an early warning for whenever you modify the source function that the test needs to be updated. 

```suggestion
test_that("output_type_id_order is unsupported (fix when supported)", {
  model_out_tbl <- readRDS(test_path("testdata", "pmf_test_model_out_tbl.rds"))
  target_observations <- readRDS(test_path("testdata", "pmf_test_target_observations.rds"))
  exp_forecast <- readRDS(test_path("testdata", "pmf_test_expected_merged_forecast.rds"))

  expect_error(
    act_forecast <- transform_pmf_model_out(
      model_out_tbl = model_out_tbl,
      target_observations = target_observations,
      output_type_id_order = "excellence"
    ),
    regexp = "unsupported"
  )
})
```

Reply:
great idea :)